07-routing-gossip.md

```diff
@@ -75,6 +75,19 @@ A recipient node:
     - MAY fail the channel.
   - if it has sent AND received a valid `announcement_signatures` message:
     - SHOULD queue the `channel_announcement` message for its peers.
+  - if it has not sent funding_locked:
+    - MAY defer handling the announcement_signatures until after it has sent funding_locked
+    - otherwise:
+      - MUST ignore it.
+
+
+### Rationale
+
+The reason for allowing deferring of a premature announcement_signatures is
+that an earlier version of the spec did not require waiting for receipt of
```

Comment:
meta: 

maybe I am still to new to protocol development. but that seems like a lot of complex overhead to make a backward compatible fix for a former bug. is that the normal way to go?

Reply:
In practice, allowing it through early turns out to be trivial (at least, in C-lightning).  In future we may stop doing that, however.  Meanwhile, usability trumps perfection...